### PR TITLE
Updated HRC20, HRC721 and HRC1155 documentation

### PR DESCRIPTION
**Pull request by j05u3**
This is only a documentation update commit, no bug fix nor features added.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/blockcoders/harmony-marketplace-sdk/blob/main/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Other... Please describe: Documentation fixes and improvements

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,9 +218,25 @@ const contract = new HRC20('0x...00', ABI, wallet, {
 
 ### Methods
 
-#### balanceOf
+#### totalSupply(): Promise&lt;BN&gt;
```
Comment (fersirni) [inline README.md:221]: Review all functions definitions as they could have an optional parameter for gas options

File: README.md [hunk 2]
```diff
@@ -212,9 +218,25 @@ const contract = new HRC20('0x...00', ABI, wallet, {
...
+
+<p>Returns the amount of tokens in existence.</p>
+
+```ts
+import { HttpProvider } from '@harmony-js/network'
+import { HRC20 } from 'harmony-marketplace-sdk'
+import * as ABI from './abi.json'
+
+// A contract instance
+const contract = new HRC20('0x...00', ABI, HttpProvider)
```
Comment (fersirni) [inline README.md:231]: This should be:
`
const wallet = new PrivateKey(
  HarmonyShards.SHARD_0,
  '45e497bd45a9049bcb649016594489ac67b9f052a6cdf5cb74ee2427a60bf25e'
)
// A contract instance
const contract = new HRC20('0x...00', ABI, wallet)
`

**Review by fersirni — CHANGES_REQUESTED** [review]
Thanks! This is a really nice improvement on our documentation. There are some conflicts to be resolved and some comments I left here. After that we can merge! cc: @0xslipk 

:+1:

File: README.md (hunk 2 above)
Comment (j05u3) [reply README.md]: Fixed in here https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/ee0e67a64f7ef634e193909de7e321080bb84c01

File: README.md (hunk 1 above)
Comment (j05u3) [reply README.md]: done here https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/c207a50c89bf3ef8378c17082fadab2286dbc4b9

let us know if there is any other comment or feedback ✌️

**Review by 0xslipk — APPROVED** [review]
LGTM, I think you still have conflicts with the main branch

**Review by fersirni — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -626,9 +717,58 @@ const contract = new HRC721('0x...00', ABI, wallet)
...
+<p>Approve or remove <code>addressOperator</code> as an operator for the caller. Operators can call <code>transferFrom</code> or <code>safeTransferFrom</code> for any token owned by the caller.<p>
+<p>Requirements
+  <ul>
+    <li>The <code>addressOperator</code> cannot be the caller.</li>
+  </ul>
+</p>
+<p>Emits an <code>ApprovalForAll</code> event.</p>
+  
+```ts
+import { HttpProvider } from '@harmony-js/network'
```
Comment (fersirni) [inline README.md:731]: remove

File: README.md [hunk 4]
```diff
@@ -626,9 +717,58 @@ const contract = new HRC721('0x...00', ABI, wallet)
...
+const tx = await contract.setApprovalForAll('0x...01', true)
+```
+
+#### isApprovedForAll(owner: string, operator: string, txOptions?: ITransactionOptions): Promise&lt;boolean&gt;
+
+<p>Returns if the <code>operator</code> is allowed to manage all of the assets of <code>owner</code>.</p>
+<p>See <code>setApprovalForAll</code></p>
+  
+```ts
+import { HttpProvider } from '@harmony-js/network'
```
Comment (fersirni) [inline README.md:753]: remove

File: README.md [hunk 5]
```diff
@@ -626,9 +717,58 @@ const contract = new HRC721('0x...00', ABI, wallet)
...
+</p>
+<p>Emits an <code>ApprovalForAll</code> event.</p>
+  
+```ts
+import { HttpProvider } from '@harmony-js/network'
+import { PrivateKey, HarmonyShards, HRC721 } from 'harmony-marketplace-sdk'
+import * as ABI from './abi.json'
+
+const wallet = new PrivateKey(
+  new HttpProvider(HarmonyShards.SHARD_0),
```
Comment (fersirni) [inline README.md:736]: replace this line for HarmonyShards.SHARD_0,

File: README.md [hunk 6]
```diff
@@ -626,9 +717,58 @@ const contract = new HRC721('0x...00', ABI, wallet)
...
+<p>Returns if the <code>operator</code> is allowed to manage all of the assets of <code>owner</code>.</p>
+<p>See <code>setApprovalForAll</code></p>
+  
+```ts
+import { HttpProvider } from '@harmony-js/network'
+import { PrivateKey, HarmonyShards, HRC721 } from 'harmony-marketplace-sdk'
+import * as ABI from './abi.json'
+
+const wallet = new PrivateKey(
+  new HttpProvider(HarmonyShards.SHARD_0),
```
Comment (fersirni) [inline README.md:758]: Same here

**Review by fersirni — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -211,9 +217,30 @@ const contract = new HRC20('0x...00', ABI, wallet, {
...
-Returns the number of tokens in owner's account.
+<p>Returns the amount of tokens in existence.</p>
+
+```ts
+import { HttpProvider } from '@harmony-js/network'
+import { PrivateKey, HarmonyShards, HRC20 } from 'harmony-marketplace-sdk'
+import * as ABI from './abi.json'
+
+const wallet = new PrivateKey(
+  new HttpProvider(HarmonyShards.SHARD_0),
```
[the remark this answers is not in the document]
Comment (j05u3) [reply README.md]: just done here: https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/276b5a749d336f82f1f92c221dd287cabe263cf5

File: README.md [hunk 8]
```diff
@@ -211,9 +217,30 @@ const contract = new HRC20('0x...00', ABI, wallet, {
...
 ### Methods
 
-#### balanceOf
+#### totalSupply(txOptions?: ITransactionOptions): Promise&lt;BN&gt;
 
-Returns the number of tokens in owner's account.
+<p>Returns the amount of tokens in existence.</p>
+
+```ts
+import { HttpProvider } from '@harmony-js/network'
```
[the remark this answers is not in the document]
Comment (j05u3) [reply README.md]: just done here: https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/276b5a749d336f82f1f92c221dd287cabe263cf5

File: README.md (hunk 3 above)
Comment (j05u3) [reply README.md]: just done here: https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/276b5a749d336f82f1f92c221dd287cabe263cf5

File: README.md (hunk 5 above)
Comment (j05u3) [reply README.md]: just done here: https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/276b5a749d336f82f1f92c221dd287cabe263cf5

File: README.md (hunk 6 above)
Comment (j05u3) [reply README.md]: just done here: https://github.com/blockcoders/harmony-marketplace-sdk/pull/54/commits/276b5a749d336f82f1f92c221dd287cabe263cf5

**Review by fersirni — APPROVED** [review]

**Pull request merged**